saleor/plugins/invoicing/__init__.py

```diff
@@ -0,0 +1,93 @@
+import os
+import re
+from datetime import datetime
+from uuid import uuid4
+
+import pytz
+from django.conf import settings
+from django.contrib.staticfiles import finders as static_finders
+from django.core.files.base import ContentFile
+from django.core.files.storage import default_storage
+from django.template.loader import get_template
+from weasyprint import HTML
+
+from ...order.models import Invoice
+
+
+def make_full_invoice_number(number=1):
+    month_and_year = datetime.now().strftime("%m/%Y")
+    return f"{number}/{month_and_year}"
+
+
+def parse_invoice_number(invoice):
+    number = re.match(r"^(\d+)\/", invoice.number).group(1)
+    return int(number)
+
+
+def generate_invoice_number():
+    last_invoice = Invoice.objects.last()
+    if not last_invoice or not last_invoice.number:
+        return make_full_invoice_number()
+
+    try:
+        last_number = parse_invoice_number(last_invoice)
+        return make_full_invoice_number(last_number + 1)
+    except (IndexError, ValueError):
+        return make_full_invoice_number()
+
+
+def chunk_products(products, product_limit):
+    """Split products to list of chunks.
+
+    Each chunk represents products per page, product_limit defines chunk size.
+    """
+    chunks = []
+    for i in range(0, len(products), product_limit):
+        limit = i + product_limit
+        chunks.append(products[i:limit])
+    return chunks
+
+
+def get_product_limit_first_page(products):
+    MAX_PRODUCTS_WITH_TABLE = 3
+    MAX_PRODUCTS_WITHOUT_TABLE = 4
```

Comment:
Maby we should move all const outside functions?

Reply:
Not my favorite approach, in typed language we would just put `const` before and leave it inside func as it's used only here and I like it more than putting it anywhere outside - though PEP8 suggest that's the way to do it so lets go with it.